### ISSUE TITLE
SilverStripe\ORM\Connect\DatabaseException could be misleading

### ISSUE THREAD

**Issue by jakxnz** [silverstripe/silverstripe-framework#8005]
## Affected Version

silverstripe/framework                    4.1.x-dev facaad5

## Description

```
class Term extends DataObject
{
    private static $table_name = 'Term';

    private static $db = [
        'Name' => 'Varchar'
    ];
}

```

```
class MyClass extends DataObject
{
    private static $has_one = [
       'Classification' => Term::class
    ];
}
```

`MyClass::get()->filter('Classification.Title', $value);`

```
[Emergency] Uncaught SilverStripe\ORM\Connect\DatabaseException: Couldn't run query: SELECT DISTINCT count(DISTINCT "MyClass"."ID") AS "Count" FROM "MyClass" LEFT JOIN "Term" AS "classification_Term" ON "classification_Term"."ID" = "MyClass"."RegionID" WHERE ("Classification"."Title" LIKE ?) 42S22-1054: Unknown column 'Classification.Title' in 'where clause'
```
Is it possible to have this read  `...WHERE ("classification_Term"."Title" LIKE ?) 42S22-1054: Unknown column 'classification_Term.Title' in 'where clause'`, if the schema has identified the relation's table?

From the error, it's hard to know if the relation is wrong, or the field. I decided to troubleshoot the relation first and it was a huge frustration; especially when I finally debugged that it was simply `Term::class` does not have a `Title` column.

I tried to work towards raising a PR, but I don't have enough context to know quite where to make the change.

**Comment by tractorcow** [comment]
This behaviour is due to SearchFilter::getDBName() not knowing where the relation is, so it just fails over to assuming you've constructed a raw query.

```php
// fallback to the provided name in the event of a joined column
// name (as the candidate class doesn't check joined records)
$parts = explode('.', $this->fullName);
return '"' . implode('"."', $parts) . '"';
```

We could fix it to throw an error (with informative message) but it would result in a lot more errors where it's ambiguous. In your case it is a real error (it would result in a sql error) but for other constructed queries, it MAY be an error, but it could just as easily not be.

The tl;dr is that we have enough information at this point to throw an error, but if we did so we'd include a lot of false-negatives as well.

At the very most, we COULD add a `trigger_error('somemessage', E_USER_NOTICE)` or `E_USER_WARNING` in this failover, so that dev environments would get both the notice, and (if it's a real error) the SQL exception as well.

@dhensby can you sanity check my evaluation please? Should we actually make a change, or just close this with "the status quo remains"?

**Comment by dhensby** [comment]
@tractorcow I think you've got it right, but I would stick with the status quo, to change it feels like it'll do more harm than good.

**Comment by jakxnz** [comment]
@tractorcow @dhensby makes sense. I feel that's fair.
